The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6119909805/job/16610750073) [(plain)](https://github.com/rust-lang/rust/commit/70060d0780d682ab8521436e15907296fec891d0/checks/16610750073/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_bf06f711-fa68-43d5-903d-71361352eefa
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=js-anonymous-functions
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_bf06f711-fa68-43d5-903d-71361352eefa
GITHUB_REF=refs/pull/115669/merge
GITHUB_REF_NAME=115669/merge
GITHUB_REF_PROTECTED=false
---
.......... (100/108)
   (108/108)


/checkout/tests/rustdoc-gui/codeblock-tooltip.goml codeblock-tooltip... FAILED
[ERROR] (around line 23 from 112): JS errors occurred: Error: TypeError: Cannot read properties of undefined (reading 'TOOLTIP_HOVER_TIMEOUT')
    at clearTooltipHoverTimeout (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:17801)
    at setTooltipHoverTimeout (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:17306)
    at e.onpointerenter (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:19322)
Error: TypeError: Cannot read properties of undefined (reading 'TOOLTIP_HOVER_TIMEOUT')
    at clearTooltipHoverTimeout (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:17801)
    at setTooltipHoverTimeout (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:17306)
    at e.onpointermove (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:19414)
/checkout/tests/rustdoc-gui/docblock-code-block-line-number.goml docblock-code-block-line-number... FAILED
[ERROR] (around line 63): JS errors occurred: Error: TypeError: Cannot read properties of undefined (reading 'id')
    at toggle.onchange (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/settings-6df3a19f92b87fa9.js:1:1431)
/checkout/tests/rustdoc-gui/notable-trait.goml notable-trait... FAILED
[ERROR] (around line 30): JS errors occurred: Error: TypeError: Cannot read properties of undefined (reading 'TOOLTIP_HOVER_TIMEOUT')
    at clearTooltipHoverTimeout (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:17801)
    at setTooltipHoverTimeout (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:17306)
    at e.onpointerenter (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:19322)
Error: TypeError: Cannot read properties of undefined (reading 'TOOLTIP_HOVER_TIMEOUT')
    at clearTooltipHoverTimeout (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:17801)
    at setTooltipHoverTimeout (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:17306)
    at e.onpointermove (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:19414)
Error: TypeError: Cannot read properties of undefined (reading 'TOOLTIP_FORCE_VISIBLE')
    at e.onclick (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/main-6909a0716c1a7e6c.js:1:18956)
/checkout/tests/rustdoc-gui/setting-auto-hide-content-large-items.goml setting-auto-hide-content-large-items... FAILED
[ERROR] (around line 29): JS errors occurred: Error: TypeError: Cannot read properties of undefined (reading 'id')
    at toggle.onchange (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/settings-6df3a19f92b87fa9.js:1:1431)
/checkout/tests/rustdoc-gui/setting-auto-hide-item-methods-docs.goml setting-auto-hide-item-methods-docs... FAILED
[ERROR] (around line 26): JS errors occurred: Error: TypeError: Cannot read properties of undefined (reading 'id')
    at toggle.onchange (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/settings-6df3a19f92b87fa9.js:1:1431)
/checkout/tests/rustdoc-gui/setting-auto-hide-trait-implementations.goml setting-auto-hide-trait-implementations... FAILED
[ERROR] (around line 25): JS errors occurred: Error: TypeError: Cannot read properties of undefined (reading 'id')
    at toggle.onchange (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/settings-6df3a19f92b87fa9.js:1:1431)
/checkout/tests/rustdoc-gui/setting-go-to-only-result.goml setting-go-to-only-result... FAILED
[ERROR] (around line 30): JS errors occurred: Error: TypeError: Cannot read properties of undefined (reading 'id')
    at toggle.onchange (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/settings-6df3a19f92b87fa9.js:1:1431)
/checkout/tests/rustdoc-gui/settings.goml settings... FAILED
[ERROR] (around line 256): JS errors occurred: Error: TypeError: Cannot read properties of undefined (reading 'id')
    at toggle.onchange (file:///checkout/obj/build/x86_64-unknown-linux-gnu/test/rustdoc-gui/doc/static.files/settings-6df3a19f92b87fa9.js:1:1431)
FFFFFFFFError: ()
  local time: Fri Sep  8 09:38:25 UTC 2023
  network time: Fri, 08 Sep 2023 09:38:25 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/librustdoc/html/static/js/main.js
+++ b/src/librustdoc/html/static/js/main.js
@@ -852,14 +852,14 @@ function preLoadCss(cssUrl) {
         window.CURRENT_TOOLTIP_ELEMENT = wrapper;
         window.CURRENT_TOOLTIP_ELEMENT.TOOLTIP_BASE = e;
         clearTooltipHoverTimeout(window.CURRENT_TOOLTIP_ELEMENT);
-        wrapper.onpointerenter = function(ev) {
+        wrapper.onpointerenter = ev => {
             // If this is a synthetic touch event, ignore it. A click event will be along shortly.
             if (ev.pointerType !== "mouse") {
                 return;
             }
             clearTooltipHoverTimeout(e);
         };
-        wrapper.onpointerleave = function(ev) {
+        wrapper.onpointerleave = ev => {
             // If this is a synthetic touch event, ignore it. A click event will be along shortly.
             if (ev.pointerType !== "mouse") {
                 return;
@@ -963,38 +963,38 @@ function preLoadCss(cssUrl) {
     }
 
     onEachLazy(document.getElementsByClassName("tooltip"), e => {
-        e.onclick = function() {
-            this.TOOLTIP_FORCE_VISIBLE = this.TOOLTIP_FORCE_VISIBLE ? false : true;
-            if (window.CURRENT_TOOLTIP_ELEMENT && !this.TOOLTIP_FORCE_VISIBLE) {
+        e.onclick = () => {
+            e.TOOLTIP_FORCE_VISIBLE = e.TOOLTIP_FORCE_VISIBLE ? false : true;
+            if (window.CURRENT_TOOLTIP_ELEMENT && !e.TOOLTIP_FORCE_VISIBLE) {
                 hideTooltip(true);
             } else {
-                showTooltip(this);
+                showTooltip(e);
                 window.CURRENT_TOOLTIP_ELEMENT.setAttribute("tabindex", "0");
                 window.CURRENT_TOOLTIP_ELEMENT.focus();
                 window.CURRENT_TOOLTIP_ELEMENT.onblur = tooltipBlurHandler;
             }
             return false;
         };
-        e.onpointerenter = function(ev) {
+        e.onpointerenter = ev => {
             // If this is a synthetic touch event, ignore it. A click event will be along shortly.
             if (ev.pointerType !== "mouse") {
                 return;
             }
-            setTooltipHoverTimeout(this, true);
+            setTooltipHoverTimeout(e, true);
         };
-        e.onpointermove = function(ev) {
+        e.onpointermove = ev => {
             // If this is a synthetic touch event, ignore it. A click event will be along shortly.
             if (ev.pointerType !== "mouse") {
                 return;
             }
-            setTooltipHoverTimeout(this, true);
+            setTooltipHoverTimeout(e, true);
         };
-        e.onpointerleave = function(ev) {
+        e.onpointerleave = ev => {
             // If this is a synthetic touch event, ignore it. A click event will be along shortly.
             if (ev.pointerType !== "mouse") {
                 return;
             }
-            if (!this.TOOLTIP_FORCE_VISIBLE &&
+            if (!e.TOOLTIP_FORCE_VISIBLE &&
                 !elemIsInParent(ev.relatedTarget, window.CURRENT_TOOLTIP_ELEMENT)) {
                 // Tooltip pointer leave gesture:
                 //
@@ -1139,7 +1139,7 @@ href="https://doc.rust-lang.org/${channel}/rustdoc/how-to-read-rustdoc.html\
      *
      * Pass "true" to reset focus for tooltip popovers.
      */
-    window.hideAllModals = function(switchFocus) {
+    window.hideAllModals = switchFocus => {
         hideSidebar();
         window.hidePopoverMenus();
         hideTooltip(switchFocus);
@@ -1148,7 +1148,7 @@ href="https://doc.rust-lang.org/${channel}/rustdoc/how-to-read-rustdoc.html\
     /**
      * Hide all the popover menus.
      */
-    window.hidePopoverMenus = function() {
+    window.hidePopoverMenus = () => {
         onEachLazy(document.querySelectorAll(".search-form .popover"), elem => {
             elem.style.display = "none";
         });

--- a/src/librustdoc/html/static/js/settings.js
+++ b/src/librustdoc/html/static/js/settings.js
@@ -59,8 +59,8 @@
             if (settingValue !== null) {
                 toggle.checked = settingValue === "true";
             }
-            toggle.onchange = function() {
-                changeSetting(this.id, this.checked);
+            toggle.onchange = () => {
+                changeSetting(toggle.id, toggle.checked);
             };
         });
         onEachLazy(settingsElement.querySelectorAll("input[type=\"radio\"]"), elem => {
@@ -224,14 +224,14 @@
 
     if (isSettingsPage) {
         // We replace the existing "onclick" callback to do nothing if clicked.
-        getSettingsButton().onclick = function(event) {
+        getSettingsButton().onclick = event => {
             event.preventDefault();
         };
     } else {
         // We replace the existing "onclick" callback.
         const settingsButton = getSettingsButton();
         const settingsMenu = document.getElementById("settings");
-        settingsButton.onclick = function(event) {
+        settingsButton.onclick = event => {
             if (elemIsInParent(event.target, settingsMenu)) {
                 return;
             }

